hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -7541,6 +7554,9 @@ void prepareGet(final Get get) throws IOException {
       () -> createRegionSpan("Region.get"));
   }
 
+  /**
+   * This method will return onheap cells, for more details, please see HBASE-26036.
```

Comment:
Nit: Only do if you make a new PR. I think you would call out that this can be an EXPENSIVE call. It may mean an extra copy from offheap to onheap buffers.

Otherwise, this looks great.

Reply:
I'll add the notice comments to where cells are copied.